### PR TITLE
Documentation generator truncates config descriptions

### LINKED ISSUE #691: Documentation generator truncates some configuration descriptions
If the configuration description contains a `-` character, the description will be truncated in the generated documentation markdown. For example, `ClassNaming`:

```kotlin
/**
 * @configuration classPattern - naming pattern (default: '[A-Z$][a-zA-Z0-9$]*')
 * @active since v1.0.0
 * @author Marvin Ramin
 */
```

Results in this documentation being generated:

image

I suppose this is due to how the description is identified based on the `-` using an hardcoded `split`ting index in `RuleVisitor`:

https://github.com/arturbosch/detekt/blob/1a22cb5bf5e4f9cce7e70f1858d4b070d6bffa75/detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt#L95-L100

### PR DESCRIPTION
**Pull request by schalkms**
This should fix #691

### CODE REVIEW

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt [hunk 1]
```diff
@@ -92,9 +92,10 @@ internal class RuleVisitor : DetektVisitor() {
 					valid
 				}
 				.map {
-					val name = it.split("-")[0].trim()
+					val delimiterIndex = it.indexOf('-')
+					val name = it.substring(0, delimiterIndex-1)
```
Comment (rock3r) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt]: Missing spaces around the `-`: `delimiterIndex-1`

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt [hunk 2]
```diff
@@ -92,9 +92,10 @@ internal class RuleVisitor : DetektVisitor() {
...
 					val defaultValue = configurationDefaultValueRegex.find(it)?.groupValues?.get(1)?.trim() ?: ""
-					val description = it.split("-")[1]
+					val description = it.substring(delimiterIndex+1)
```
Comment (rock3r) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt]: Missing spaces around the `+`: `delimiterIndex+1`

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt (hunk 1 above)
Comment (schalkms) [reply detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt]: There is a trim(). It doesn't matter.

Comment (rock3r) [reply detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/collection/RuleVisitor.kt]: I'm talking about your code at this line, not the documentation it generates. The code needs a reformatting.

**Pull request merged**